src/app/modules/twilio/twilio.controller.ts

```diff
@@ -0,0 +1,33 @@
+import { StatusCodes } from 'http-status-codes'
+import Twilio from 'twilio'
+
+import { ITwilioSmsWebhookBody } from 'src/types/twilio'
+
+import { ControllerHandler } from '../core/core.types'
+
+import { logFailedSmsDelivery } from './twilio.service'
+
+const smsDeliveryFailedStatus = ['undelivered', 'failed']
+
+/**
+ * Middleware which validates that a request came from Twilio Webhook
+ */
+const validateTwilioWebhook = Twilio.webhook()
+
+/**
+ * Process the Webhook requests if they are failed or unsuccessful, ignoring the rest
+ *
+ * @param req Express request object
+ * @param res - Express response object
+ */
+export const twilioSmsUpdates: ControllerHandler<
+  unknown,
+  never,
+  ITwilioSmsWebhookBody
+> = async (req, res) => {
+  if (smsDeliveryFailedStatus.includes(req.body.MessageStatus))
+    logFailedSmsDelivery(req.body)
+  res.sendStatus(StatusCodes.OK)
```

Comment:
should we add the `return` keyword here? We've had cases of multiple `res.sendXXX` calls in the past due to forgetting to return

Reply:
How does the multple res.sendxxx call happen? Can't really visualize it in my head :sweat_smile:

Reply:
```ts
const someFn = () => {
  if (someCondition) {
    res.sendStatus(404)
  }
  // ... a bunch of extra work that happened because a return was not called
  res.sendStatus(200) // Double send error will occur here
}
```

Not likely in your code, but still good to be consistent across the codebase